netutils/route.py

```diff
@@ -43,3 +46,159 @@ def longest_prefix_match(ip_addr, routes):
         return str(sorted(networks)[-1])
     except IndexError as error:
         raise NoRouteFound("No Matching Route Found.") from error
+
+
+IPNetwork = Union[ipaddress.IPv4Network, ipaddress.IPv6Network]
+
+
+# pylint: disable=too-many-branches
+def prefix_aggregate(
```

Comment:
I think this can be simplified, certainly not complete, but should be able to get the idea.

```
import ipaddress

floor = 340282366920938463463374607431768211455
ceiling = 0
prefix = 128

networks = ["10.0.0.0/24", "10.0.1.0/24", "10.0.10.0/24"]

for net in networks:
  ip_obj = ipaddress.ip_network(net)
  net_floor = int(ip_obj.network_address)
  net_ceiling = int(ip_obj.broadcast_address)
  if net_floor < floor:
    floor = net_floor
  if net_ceiling > ceiling:
    ceiling = net_ceiling
  if ip_obj.prefixlen < prefix:
    prefix = ip_obj.prefixlen

for _prefix in range(prefix, 0, -1):
  ip_obj = ipaddress.ip_interface(f"{str(ipaddress.ip_address(floor))}/{_prefix}")
  if int(ip_obj.network.network_address) <= floor and int(ip_obj.network.broadcast_address) >= ceiling:
    print(ip_obj.network)
    break
```

Reply:
let me work on this

Reply:
@viktorkertesz any luck?